### PR TITLE
Fixes #280: redacted email-filter & object conditions with new operators

### LINKED ISSUE #280: Update supported operators for input fields in conditions
* Misc » Object conditions » Basic object conditions » Text field (input)
  * Supported operators now are:
    * `contains` - matches if text contains a specific string
    * `contains not` - matches if text does not contain a specific string
    * `is any of` - matches if text is equal to any of given tokens*
    * `is none of` - matches if text is not equal to all of given tokens*
    * `starts with one of` - matches if text starts with one of given tokens*
    * `ends with one of` - matches if text ends with one of given tokens*
    * `matches regex` - evaluates if text matches provided regular expression**
    *  `does not match regex` - evaluates if text does not match provided regular expression**
* Docs » Email » Filters
  * Instead of `contains` operator and a value with`regex:...` prefix, regular expressions are available now via dedicated operators `matches regex` and `does not match regex`. Adjust the examples and modify the paragraph near the end to describe the new behavior.

*Note: tokens work very much in the same way as tags, but they don't support a comma (`,`) as a delimiter, while tags do. Not sure if this should be mentioned, just as extra info.

**Additional note: `matches regex` and `does not match regex` are supported only in Triggers, Time Accounting selector, Postmaster Filters, Automatic ticket assignment and Core Workflow, due to performance concerns. Again, not sure if this should be signaled somehow to the admin in the docs, just as extra info.

### PR DESCRIPTION
**Pull request by ralf401**

### CODE REVIEW

**Comment by ralf401** [comment]
@dvuckovic I used your wording for the operators from the issue because I think it is crisp and clear. Hope you don't mind.
I also restructured the filter page a little bit.

File: channels/email/filters.rst [hunk 1]
```diff
@@ -17,33 +17,54 @@ will be applied to the ticket that is created or updated by this email.
...
+the following actions can be combined. The supported matches are:
+
+   * Contains
+   * Contains not
+   * Is any of
+   * Is none of
+   * Starts with one of
+   * Ends with one of
+   * Matches regex
+   * Does not match regex
```
Comment (dvuckovic) [inline channels/email/filters.rst:32]: I recommend we style the list of actual operators like in other places: lowercase and in *italics*. Something like this:

```rst
   * *contains*
   * *contains not*
   * *is any of*
   * *is none of*
   * ...
```

File: channels/email/filters.rst [hunk 2]
```diff
@@ -17,33 +17,54 @@ will be applied to the ticket that is created or updated by this email.
...
-
-    Note that the Group action only has an effect when the matching email
-    results in a new ticket. Zammad will not change the group of existing
-    tickets.
+Automatically dispatch tickets to responsible staff based on organization name:
+  | Organization: Starts with one of: ``A`` ``B`` ``C``
+  | Owner: Emily Adams
 
 Automatically increase the priority of tickets from a VIP customer:
   | From: contains: ``ourvipcustomer@example.com``
```
Comment (dvuckovic) [inline channels/email/filters.rst:65]: `contains:` => `*contains:*`

File: channels/email/filters.rst [hunk 3]
```diff
@@ -17,33 +17,54 @@ will be applied to the ticket that is created or updated by this email.
 
 Zammad comes with system filters as well. While you can't change them, it might
 be useful for you what they actually do. Learn more on
-:doc:`filters/system-filters`
+:doc:`filters/system-filters`.
+
+Different attributes of a filter can be combined with each other. Likewise,
+the following actions can be combined. The supported matches are:
```
[the remark this answers is not in the document]
Comment (dvuckovic) [reply channels/email/filters.rst:23]: The attributes were previously refered as actions in this document, but this is wrong. We are *matching* provided `values` on object __atributes__, so we should be consistent. If you see any other places with the confusing terminology, feel free to improve them.

File: channels/email/filters.rst [hunk 4]
```diff
@@ -17,33 +17,54 @@ will be applied to the ticket that is created or updated by this email.
...
   | Priority: 3 high
 
-  .. note::
-
-    Note that the Priority action only has an effect when the matching email
-    results in a new ticket. Zammad will not change the priority of existing
-    tickets.
-
 Automatically tag and close spam tickets that have been marked as spam by anexternal spam filter (e.g. SpamAssassin):
   | X-Spam-Flag: contains: ``YES``
```
Comment (dvuckovic) [inline channels/email/filters.rst:69]: `contains:` => `*contains:*`

File: channels/email/filters.rst [hunk 5]
```diff
@@ -52,27 +73,8 @@ Automatically tag and close spam tickets that have been marked as spam by anexte
...
-the following actions can be combined. The supported matches are "contains"
-and "contains not"; for advanced matching, you can use regular expressions by
-prefixing the string with ``regex:``.
-
-Note that Zammad matches against the full header, e.g. for a mail with
-``From: Display Name <display.name@example.com>``, the ``From`` condition will
-test against ``Display Name <display.name@example.com>``. This is especially
-important when using anchored regular expressions;
-``regex:^display\.name@example.com$`` would not match this mail!
+    results in a new ticket. It will add the tag though if it is missing, even
```
Comment (dvuckovic) [inline channels/email/filters.rst:76]: `State action` => `State attribute`

File: misc/object-conditions/basics.rst [hunk 6]
```diff
@@ -341,10 +341,57 @@ Text field (input)
...
-   The configured "Type" of input fields as no impact on the available options.
+   The configured "Type" of input fields has no impact on the available options.
 
-| Check if any field of type text contains a specific string (or not)?
-| Available matching: *contains*, *contains not* or *has changed*
+Check if any field of type text contains a specific string (or not)?
+
+Available operators for matching:
+
+   Contains
```
Comment (dvuckovic) [inline misc/object-conditions/basics.rst:350]: Lowercase + *italics*, please. For all operators listed below.

File: misc/object-conditions/basics.rst [hunk 7]
```diff
@@ -341,10 +341,57 @@ Text field (input)
...
+   Matches regex
+      Evaluates if text matches provided regular expression.
+
+   Does not match regex
+      Evaluates if text does not match provided regular expression.
+
+.. hint::
+   **Regex support**
+
+   ``matches regex`` and ``does not match regex`` are supported only in
```
Comment (dvuckovic) [inline misc/object-conditions/basics.rst:377]: Operator style should be in *italics*, instead of `code`.

File: misc/object-conditions/basics.rst [hunk 8]
```diff
@@ -341,10 +341,57 @@ Text field (input)
...
+   Please note, that the input field for tokens doesn't support the comma as
+   separator (as in the input field for tags). If you use the comma in the
+   token input field, the comma is included in you token.
+
+   **Example**:
+
+   .. figure:: /images/misc/object-conditions/operator-token-input.png
+      :alt: Example of comma usage in token input field
+
+      ``A`` added with enter/tab, ``B`` and ``C`` separted with comma (resulting
```
Comment (dvuckovic) [inline misc/object-conditions/basics.rst:393]: `separted` => `separated`

**Review by dvuckovic — CHANGES_REQUESTED** [review]

**Review by dvuckovic — APPROVED** [review]

**Pull request merged**